### ISSUE TITLE
Doom update error vterm

### ISSUE THREAD

**Issue by ratnadeep007** [hlissner/doom-emacs#2988]
**What did you expect to happen?**
Run `doom update` without error after adding `vterm`

**What actually happened?**
```
➜ ~/.emacs.d/bin/doom update
> (Re)generating core autoloads...
  > Generating core autoloads...
  > Byte-compiling core autoloads file...
  ✓ Generated .local/autoloads.elc
> Updating packages (this may take a while)...
  ! (84/171) Encountered error with vterm
    x (error Failed to run "git"; see buffer *straight-process*)
      - fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
      Use '--' to separate paths from revisions, like this:
      'git <command> [<revision>...] -- [<file>...]'
      HEAD

x Encountered 1 error(s), the offending packages: vterm
✓ All 171 packages are up-to-date
✓ Finished! (61.7180s)
```

**Steps to reproduce:**
1. Running on macOS uncommented `vterm`
2. Ran `doom update` and it throws above error
3. `doom refresh` ran successfully
4. When tried to open terminal it says `Cannot open file: No such file or directory, vterm`

**System information:**
<details><pre>
➜ ~/.emacs.d/bin/doom info
((emacs
  (version . "26.3")
  (features . "JPEG RSVG IMAGEMAGICK GLIB NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS LCMS2")
  (build . "Aug 30, 2019")
  (buildopts "--disable-dependency-tracking --disable-silent-rules --enable-locallisppath=/usr/local/share/emacs/site-lisp --infodir=/usr/local/Cellar/emacs-plus/26.3/share/info/emacs --prefix=/usr/local/Cellar/emacs-plus/26.3 --with-xml2 --without-dbus --with-gnutls --with-imagemagick --with-modules --with-rsvg --with-ns --disable-ns-self-contained")
  (windowsys . batch)
  (daemonp))
 (doom
  (version . "2.0.9")
  (build . "HEAD -> develop, origin/develop, origin/HEAD f82771980 2020-03-11 23:20:06 -0400")
  (dir . "~/.doom.d/"))
 (system
  (type . darwin)
  (config . "x86_64-apple-darwin18.5.0")
  (shell . "/usr/local/bin/fish")
  (uname . "Darwin 18.7.0 Darwin Kernel Version 18.7.0: Thu Jan 23 06:52:12 PST 2020; root:xnu-4903.278.25~1/RELEASE_X86_64 x86_64")
  (path "~/go/bin" "~/.local/bin" "~/Library/Python/3.7/bin" "/usr/local/opt/libpq/bin" "~/.poetry/bin" "~/.config/nvm/12.16.1/bin" "~/.wasmer/bin" "~/.wasmtime/bin" "~/.cargo/bin" "/usr/local/sbin" "~/bin" "~/.cargo/bin" "~/softwares/flutter/bin" "/usr/local/bin" "/usr/bin" "/bin" "/usr/sbin" "/sbin" "/usr/local/go/bin" "~/.wasmer/globals/wapm_packages/.bin" "/usr/local/Cellar/emacs-plus/26.3/libexec/emacs/26.3/x86_64-apple-darwin18.5.0"))
 (config
  (envfile . envvar-file)
  (elc-files . 0)
  (modules :completion company ivy :ui doom doom-dashboard doom-quit hl-todo modeline nav-flash ophints (popup +defaults) treemacs vc-gutter vi-tilde-fringe window-select workspaces :editor (evil +everywhere) file-templates fold snippets :emacs dired electric vc :term vterm :checkers syntax :tools (eval +overlay) lookup magit :lang data emacs-lisp markdown (org +dragndrop +present) sh :config (default +bindings +smartparens))
  (packages "n/a")
  (unpin "n/a")
  (elpa "n/a")))
</pre></details>

**Comment by hlissner** [comment]
Does this fix your issue?

```sh
rm -rf ~/.emacs.d/.local/straight/repos/straight.el
rm -rf ~/.emacs.d/.local/straight/repos/vterm
doom sync
```

**Comment by ratnadeep007** [comment]
Yeah it solved the issue but running update again throws same error.
Also even after `doom sync` is successful and it see vterm build is running successful on `SPC-o-t` it throws error `Cannot open file: No such file or directory, vterm`

**Comment by hlissner** [comment]
This usually means the build process failed. Press `SPC b B` and look for a buried vterm or compilation buffer. It should tell you why it failed.

**Comment by ratnadeep007** [comment]
I only one vterm related buffer `doom:vterm-popup:main` and its completely blank

**Comment by nagallego** [comment]
Is there another wat to solve it? Same error here but the commands mentions above doesn't solve the issues for me
